### PR TITLE
feat: Session management

### LINKED ISSUE #23: Session management
### Done criteria

  - Define session interface
  - Implement in-memory session repository
  - Implement session service that creates new sessions ( generates id, etc. )

Depends on #37

### PR DESCRIPTION
**Pull request by elementbound**
Closes #23

### CODE REVIEW

File: packages/server/src/session/session.module.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Module } from '@nestjs/common';
+import { SessionService } from './session.service';
+import { SessionController } from './session.controller';
+import { SessionDAO } from './session.dao';
+import { IDGenerator } from 'src/id.generator';
+
+@Module({
+  providers: [SessionService, SessionDAO, IDGenerator],
```
Comment (elementbound) [inline packages/server/src/session/session.module.ts:8]: Will move the IDGenerator to a utility module

File: packages/server/src/assert.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export function assert(condition: boolean, message: string = ''): void {
```
Comment (ocsi0520) [inline packages/server/src/assert.ts:1]: server-side we can use `node:assert`
https://bun.sh/docs/runtime/nodejs-apis

File: packages/server/tsconfig.build.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
```
Comment (ocsi0520) [inline packages/server/tsconfig.build.json:1]: I got that, that we have a separate tsconfig only for build, where we can exclude the type-checks on
node_modules, test, dist, and test files, but where do we reference it?

**Review by ocsi0520 — CHANGES_REQUESTED** [review]

File: packages/server/src/assert.ts (hunk 2 above)
Comment (elementbound) [reply packages/server/src/assert.ts]: Done.

File: packages/server/tsconfig.build.json (hunk 3 above)
Comment (elementbound) [reply packages/server/tsconfig.build.json]: IIIRC it's a leftover from nest's starter template. Removed, since we don't actually transpile server code.

**Pull request merged**